macros/grant_schema_access.sql

```diff
@@ -0,0 +1,44 @@
+{% macro grant_schema_access(roles) %}
+
+{% if execute %}
+
+{% if roles is not string and roles is not iterable %}
+  {% do exceptions.raise_compiler_error('"roles" must be a string or a list') %}
+{% elif roles is string %}
+  {% set roles = [roles] %}
+{% endif %}
+
+{% set models = graph.nodes.values() | selectattr("resource_type", "equalto", "model") | list %}
```

Comment:
```suggestion
{% set models = graph.nodes.values() | selectattr("resource_type", "in", ["model", "seed"]) | list %}
```

Reply:
available resources are `['model', 'snapshot', 'test', 'seed', 'operation']` should we use this as an argument with a default of `["model", "seed"]` in case someone there's a need for other types?

Reply:
`snapshot` yes but i dont think the others have schema to grant access to or are testable from metaplane's perspective